ui/src/app/components/object-detail/object-detail.component.ts

```diff
@@ -720,20 +782,22 @@ export class ObjectDetailComponent implements OnInit {
     })
   }
   dropIndexKey(index: number) {
-    for (let i = 0; i < this.indexData.length; i++) {
-      if (i === index || this.indexData[i].spColName === undefined) {
-        this.indexData.splice(index, 1)
-      }
+    if (this.localIndexData[index].spColName) {
+      this.localIndexData[index].spColName = undefined
```

Comment:
rather than setting it to undefined is there no way to delete the property ?

Reply:
Set to '' for now. If deleted entirely Source column in index will also go. Some changes will be coming here after MS5 gets merged. There we have altered the datasources for tables a little bit.